### DISCUSSION TITLE
Error: Cannot access a disposed object. Object name: 'DatabaseConnectionState'.

### DISCUSSION THREAD

**Discussion by georgiosd** [ravendb/ravendb#15459] [General]
Hey everyone

We're seeing an error bubble up to `UnobservedTaskException` with no stack trace. There's only Raven on this project so I'm guessing that's where it's coming from.

```
System.ObjectDisposedException
Cannot access a disposed object. Object name: 'DatabaseConnectionState'.
```

Any ideas how to debug it? It's very low occurrence (4 times in 30 days) but would still be great to resolve it.

Thank you

**Comment by ayende** [comment]
This is part of the `Changes()` API, and means that you disposed the connection but are still using it?
Relevant code is:

https://github.com/ravendb/ravendb/blob/105e921b2690fb73b41039568cc3222d2a59feca/src/Raven.Client/Documents/Changes/DatabaseConnectionState.cs#L97

**Reply by georgiosd** [reply]
That's really strange. We don't use the Changes API. Do subscriptions use it under the hood?

**Reply by ayende** [reply]
are you using aggressive caching?

**Comment by georgiosd** [comment]
Yes, in some places, indeed. But the scope is disposed at the usage:

using(aggressive caching)
{
   doc = load()
}

**Reply by ayende** [reply]
That is using changes under the cover.
How do you observe this exception?

**Reply by georgiosd** [reply]
It bubbled up to Sentry through their `UnobservedTaskException` handler so it looks like you're not handling it somewhere?

**Reply by ayende** [reply]
That isn't actually raised. It should be handled internally.
It it possible that the issue is that it is handled in the `~Task` finalizer, I guess.
We aren't observing the exception, after all.

**Comment by georgiosd** [comment]
That would make sense, thank you.
